File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        set<int>s;
        for(int x : nums){
            s.insert(x);
        }
        int i = 1;
        for(auto it = s.begin(); it != s.end(); it++){
            while(it != s.end() and *it != i){
                it++;
            }
            if(it == s.end()){
                return i;
            }
            if(*it != i){
                break;
            }
            ++i;
        }
        return i;
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        int n = nums.length;
        boolean[] seen = new boolean[n + 1];
        
        for(int num : nums){
            if(num >= 1 && num <= n)    seen[num] = true;
        }
        
        for(int i = 1; i < n + 1; i++){
            if(!seen[i])    return i;
        }
        
        return n + 1;
    }
}